js/src/components/enhanced-conversion-tracking-settings/cta.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { ENHANCED_ADS_CONVERSION_STATUS } from '.~/constants';
+import { useAppDispatch } from '.~/data';
+import AppButton from '.~/components/app-button';
+import AcceptTerms from './accept-terms';
+import useAcceptedCustomerDataTerms from '.~/hooks/useAcceptedCustomerDataTerms';
+import useAllowEnhancedConversions from '.~/hooks/useAllowEnhancedConversions';
+import useTermsPolling from './useTermsPolling';
+
+const CTA = ( {
+	disableLabel = __( 'Disable', 'google-listings-and-ads' ),
+	enableLabel = __( 'Enable', 'google-listings-and-ads' ),
+	onEnableClick = noop,
+	onDisableClick = noop,
+} ) => {
+	const [ startBackgroundPoll, setStartBackgroundPoll ] = useState( false );
+	const { updateEnhancedAdsConversionStatus } = useAppDispatch();
+	const { acceptedCustomerDataTerms } = useAcceptedCustomerDataTerms();
+	const { allowEnhancedConversions } = useAllowEnhancedConversions();
+	useTermsPolling( startBackgroundPoll );
+
+	const handleDisable = useCallback( () => {
+		if ( ! acceptedCustomerDataTerms ) {
+			return;
+		}
+
+		updateEnhancedAdsConversionStatus(
+			ENHANCED_ADS_CONVERSION_STATUS.DISABLED
+		);
+
+		onDisableClick();
+	}, [
+		updateEnhancedAdsConversionStatus,
+		acceptedCustomerDataTerms,
+		onDisableClick,
+	] );
+
+	// Turn off polling when the user has accepted the terms.
+	useEffect( () => {
+		if ( acceptedCustomerDataTerms && startBackgroundPoll ) {
+			setStartBackgroundPoll( false );
+		}
+	}, [ acceptedCustomerDataTerms, startBackgroundPoll ] );
```

Comment:
Using another state `startBackgroundPoll` and creating another layer hook `useTermsPolling` to serve all the data polling processing for the accepted status of customer data terms seems too overly complex.

I think the same results could be achieved in a more simple way:
1. Move the required condition check from `useTermsPolling` to `useAutoCheckEnhancedConversionTOS`
2. Move the determinations of starting and stopping the polling to `useAutoCheckEnhancedConversionTOS`
3. Return the `isPolling` from `useAutoCheckEnhancedConversionTOS`
4. Call the `useAutoCheckEnhancedConversionTOS` hook in `EnhancedConversionTrackingSettings` instead of `AcceptTerms`

Combining the consideration in https://github.com/woocommerce/google-listings-and-ads/pull/2242#issuecomment-2021037071:

> 2\. It is possible for the EC modal to be displayed in an enabled sate ([comment](https://github.com/woocommerce/google-listings-and-ads/pull/2242#pullrequestreview-1905425020))
> 
> This is still technically true, but we are now making sure the previous options are removed when an Ads account is disconnected ([here](https://github.com/woocommerce/google-listings-and-ads/pull/2242/files#diff-4352e33eec496eff04435ed74e2520968d0f5adc00dff087ca5cc9660b79d4e8R221-R222)). This fixes the situation you mentioned where disconnecting accounts and going back through the setup process could result in a user seeing this state.
> 
> Now, the only way that I think a user could end in this state is if the options failed to delete for some reason, or if they got set manually outside of the expected onboarding process. This all seems pretty edge case to me, so we've not added an additional check to disable the second page of the modal if paid ads are already set up AND EC has already been enabled.

The relevant code could be simpler, for example, I prepared a git diff result based on the revision a3fb3e65ad5f268f6540b02d33194db1983cb0c5:

```diff
diff --git a/js/src/components/enhanced-conversion-tracking-settings/accept-terms.js b/js/src/components/enhanced-conversion-tracking-settings/accept-terms.js
index 832274b94..9c5a1799f 100644
--- a/js/src/components/enhanced-conversion-tracking-settings/accept-terms.js
+++ b/js/src/components/enhanced-conversion-tracking-settings/accept-terms.js
@@ -2,7 +2,6 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { noop } from 'lodash';
 import { useCallback } from '@wordpress/element';
 
 /**
@@ -18,7 +17,6 @@ const AcceptTerms = ( {
 		'Accept Terms & Conditions',
 		'google-listings-and-ads'
 	),
-	onAcceptTerms = noop,
 } ) => {
 	const { url } = useGoogleAdsEnhancedConversionTermsURL();
 	const { updateEnhancedAdsConversionStatus } = useAppDispatch();
@@ -31,9 +29,8 @@ const AcceptTerms = ( {
 			updateEnhancedAdsConversionStatus(
 				ENHANCED_ADS_CONVERSION_STATUS.PENDING
 			);
-			onAcceptTerms();
 		},
-		[ updateEnhancedAdsConversionStatus, url, onAcceptTerms ]
+		[ updateEnhancedAdsConversionStatus, url ]
 	);
 
 	return (
diff --git a/js/src/components/enhanced-conversion-tracking-settings/cta.js b/js/src/components/enhanced-conversion-tracking-settings/cta.js
index dc2ff94ad..897316d72 100644
--- a/js/src/components/enhanced-conversion-tracking-settings/cta.js
+++ b/js/src/components/enhanced-conversion-tracking-settings/cta.js
@@ -3,7 +3,7 @@
  */
 import { noop } from 'lodash';
 import { __ } from '@wordpress/i18n';
-import { useCallback, useState, useEffect } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
@@ -13,43 +13,16 @@ import { useAppDispatch } from '.~/data';
 import AppButton from '.~/components/app-button';
 import AcceptTerms from './accept-terms';
 import useAcceptedCustomerDataTerms from '.~/hooks/useAcceptedCustomerDataTerms';
-import useAllowEnhancedConversions from '.~/hooks/useAllowEnhancedConversions';
-import useTermsPolling from './useTermsPolling';
+import useAutoCheckEnhancedConversionTOS from '.~/hooks/useAutoCheckEnhancedConversionTOS';
 
 const CTA = ( {
-	disableLabel = __( 'Disable', 'google-listings-and-ads' ),
 	enableLabel = __( 'Enable', 'google-listings-and-ads' ),
 	onEnableClick = noop,
-	onDisableClick = noop,
 } ) => {
-	const [ startBackgroundPoll, setStartBackgroundPoll ] = useState( false );
 	const { updateEnhancedAdsConversionStatus } = useAppDispatch();
 	const { acceptedCustomerDataTerms } = useAcceptedCustomerDataTerms();
-	const { allowEnhancedConversions } = useAllowEnhancedConversions();
-	useTermsPolling( startBackgroundPoll );
 
-	const handleDisable = useCallback( () => {
-		if ( ! acceptedCustomerDataTerms ) {
-			return;
-		}
-
-		updateEnhancedAdsConversionStatus(
-			ENHANCED_ADS_CONVERSION_STATUS.DISABLED
-		);
-
-		onDisableClick();
-	}, [
-		updateEnhancedAdsConversionStatus,
-		acceptedCustomerDataTerms,
-		onDisableClick,
-	] );
-
-	// Turn off polling when the user has accepted the terms.
-	useEffect( () => {
-		if ( acceptedCustomerDataTerms && startBackgroundPoll ) {
-			setStartBackgroundPoll( false );
-		}
-	}, [ acceptedCustomerDataTerms, startBackgroundPoll ] );
+	const isPolling = useAutoCheckEnhancedConversionTOS();
 
 	const handleEnable = useCallback( () => {
 		if ( ! acceptedCustomerDataTerms ) {
@@ -67,23 +40,18 @@ const CTA = ( {
 		onEnableClick,
 	] );
 
-	const handleOnAcceptTerms = () => {
-		setStartBackgroundPoll( true );
-	};
-
-	if ( startBackgroundPoll ) {
-		return <AppButton isSecondary disabled loading />;
+	if ( isPolling ) {
+		return <AppButton isSecondary loading />;
 	}
 
 	if ( ! acceptedCustomerDataTerms ) {
-		return <AcceptTerms onAcceptTerms={ handleOnAcceptTerms } />;
-	}
-
-	if ( allowEnhancedConversions === ENHANCED_ADS_CONVERSION_STATUS.ENABLED ) {
 		return (
-			<AppButton isPrimary isDestructive onClick={ handleDisable }>
-				{ disableLabel }
-			</AppButton>
+			<AcceptTerms
+				acceptTermsLabel={ __(
+					'Sign terms of service on Google Ads',
+					'google-listings-and-ads'
+				) }
+			/>
 		);
 	}
 
diff --git a/js/src/components/enhanced-conversion-tracking-settings/index.js b/js/src/components/enhanced-conversion-tracking-settings/index.js
index f9f95bcfc..5886eba87 100644
--- a/js/src/components/enhanced-conversion-tracking-settings/index.js
+++ b/js/src/components/enhanced-conversion-tracking-settings/index.js
@@ -12,6 +12,7 @@ import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
 import Section from '.~/wcdl/section';
 import PendingNotice from '.~/components/enhanced-conversion-tracking-settings/pending-notice';
 import useAllowEnhancedConversions from '.~/hooks/useAllowEnhancedConversions';
+import useAutoCheckEnhancedConversionTOS from '.~/hooks/useAutoCheckEnhancedConversionTOS';
 import Toggle from './toggle';
 import AcceptTerms from './accept-terms';
 
@@ -34,6 +35,8 @@ const EnhancedConversionTrackingSettings = () => {
 	const { acceptedCustomerDataTerms } = useAcceptedCustomerDataTerms();
 	const { allowEnhancedConversions } = useAllowEnhancedConversions();
 
+	useAutoCheckEnhancedConversionTOS();
+
 	// @todo: Remove condition once R1 PRs are merged since there should always be a connected Ads account.
 	if ( ! googleAdsAccount || ! googleAdsAccount.id ) {
 		return null;
diff --git a/js/src/components/enhanced-conversion-tracking-settings/pending-notice.js b/js/src/components/enhanced-conversion-tracking-settings/pending-notice.js
index 1fb49e8b8..b2839e4f9 100644
--- a/js/src/components/enhanced-conversion-tracking-settings/pending-notice.js
+++ b/js/src/components/enhanced-conversion-tracking-settings/pending-notice.js
@@ -10,13 +10,11 @@ import { createInterpolateElement, useCallback } from '@wordpress/element';
 import { ENHANCED_ADS_CONVERSION_STATUS } from '.~/constants';
 import { useAppDispatch } from '.~/data';
 import TrackableLink from '.~/components/trackable-link';
-import useTermsPolling from './useTermsPolling';
 import useGoogleAdsEnhancedConversionTermsURL from '.~/hooks/useGoogleAdsTermsURL';
 
 const PendingNotice = () => {
 	const { updateEnhancedAdsConversionStatus } = useAppDispatch();
 	const { url } = useGoogleAdsEnhancedConversionTermsURL();
-	useTermsPolling();
 
 	const handleOnClick = useCallback(
 		( event ) => {
diff --git a/js/src/hooks/useAutoCheckEnhancedConversionTOS.js b/js/src/hooks/useAutoCheckEnhancedConversionTOS.js
index 1a576165c..00d2a5881 100644
--- a/js/src/hooks/useAutoCheckEnhancedConversionTOS.js
+++ b/js/src/hooks/useAutoCheckEnhancedConversionTOS.js
@@ -1,25 +1,25 @@
 /**
  * External dependencies
  */
-import { useCallback, useState } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { ENHANCED_ADS_CONVERSION_STATUS } from '.~/constants';
 import { useAppDispatch } from '.~/data';
 import useWindowFocusCallbackIntervalEffect from '.~/hooks/useWindowFocusCallbackIntervalEffect';
+import useAcceptedCustomerDataTerms from '.~/hooks/useAcceptedCustomerDataTerms';
+import useAllowEnhancedConversions from '.~/hooks/useAllowEnhancedConversions';
 
 const useAutoCheckEnhancedConversionTOS = () => {
-	const [ polling, setPolling ] = useState( false );
+	const { acceptedCustomerDataTerms } = useAcceptedCustomerDataTerms();
+	const { allowEnhancedConversions } = useAllowEnhancedConversions();
 	const { fetchAcceptedCustomerDataTerms } = useAppDispatch();
 
-	const startEnhancedConversionTOSPolling = useCallback( () => {
-		setPolling( true );
-	}, [] );
-
-	const stopEnhancedConversionTOSPolling = useCallback( () => {
-		setPolling( false );
-	}, [] );
+	const polling =
+		! acceptedCustomerDataTerms &&
+		allowEnhancedConversions === ENHANCED_ADS_CONVERSION_STATUS.PENDING;
 
 	const checkStatus = useCallback( async () => {
 		if ( ! polling ) {
@@ -31,10 +31,7 @@ const useAutoCheckEnhancedConversionTOS = () => {
 
 	useWindowFocusCallbackIntervalEffect( checkStatus, 30 );
 
-	return {
-		startEnhancedConversionTOSPolling,
-		stopEnhancedConversionTOSPolling,
-	};
+	return polling;
 };
 
 export default useAutoCheckEnhancedConversionTOS;
diff --git a/js/src/product-feed/submission-success-guide/enhanced-conversion/actions.js b/js/src/product-feed/submission-success-guide/enhanced-conversion/actions.js
index fca858d63..d1e2e23ff 100644
--- a/js/src/product-feed/submission-success-guide/enhanced-conversion/actions.js
+++ b/js/src/product-feed/submission-success-guide/enhanced-conversion/actions.js
@@ -15,7 +15,7 @@ import CTA from '.~/components/enhanced-conversion-tracking-settings/cta';
 const Actions = ( { onModalClose = noop } ) => {
 	const { createNotice } = useDispatchCoreNotices();
 
-	const handleEnableOrDisableClick = useCallback( () => {
+	const handleEnableClick = useCallback( () => {
 		createNotice(
 			'info',
 			__( 'Status successfully set', 'google-listings-and-ads' )
@@ -38,12 +38,7 @@ const Actions = ( { onModalClose = noop } ) => {
 			</AppButton>
 
 			<CTA
-				onEnableClick={ handleEnableOrDisableClick }
-				onDisableClick={ handleEnableOrDisableClick }
-				acceptTermsLabel={ __(
-					'Sign terms of service on Google Ads',
-					'google-listings-and-ads'
-				) }
+				onEnableClick={ handleEnableClick }
 				enableLabel={ __( 'Confirm', 'google-listings-and-ads' ) }
 			/>
 		</Fragment>

```
(Please note that the diff result only contains the minimal adjustments to present the idea)

Reply:
Thanks for the suggestion and for confirming that not addressing the edge case is not a blocker.